ddprof-test/src/test/java/com/datadoghq/profiler/cpu/CTimerSamplerTest.java

```diff
@@ -0,0 +1,61 @@
+package com.datadoghq.profiler.cpu;
+
+import com.datadoghq.profiler.AbstractProfilerTest;
+import com.datadoghq.profiler.Platform;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junitpioneer.jupiter.RetryingTest;
+import org.openjdk.jmc.common.item.IItem;
+import org.openjdk.jmc.common.item.IItemCollection;
+import org.openjdk.jmc.common.item.IItemIterable;
+import org.openjdk.jmc.common.item.IMemberAccessor;
+import org.openjdk.jmc.common.unit.IQuantity;
+import org.openjdk.jmc.flightrecorder.jdk.JdkAttributes;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class CTimerSamplerTest extends AbstractProfilerTest {
+
+    private ProfiledCode profiledCode;
+
+    @Override
+    protected void before() {
+        profiledCode = new ProfiledCode(profiler);
+    }
+
+    @RetryingTest(10)
+    public void test() throws ExecutionException, InterruptedException {
+        for (int i = 0, id = 1; i < 100; i++, id += 3) {
+            profiledCode.method1(id);
+        }
+        stopProfiler();
+        IItemCollection events = verifyEvents("datadog.ExecutionSample");
+
+        // on mac the usage of itimer to drive the sampling provides very unreliable outputs
```

Comment:
this comment is probably wrong - what is the status of this timer on mac?

Reply:
Good catch. The whole test should be skipped on anything but LInux.